src/Microsoft.DotNet.Tools.Compiler/Program.cs

```diff
@@ -827,4 +878,34 @@ private static string EnsureTrailingCharacter(string path, char trailingCharacte
             return path + trailingCharacter;
         }
     }
+
+    class Preconditions
```

Comment:
Initially I started with 3 subclasses, then I realized there's no subtype specific behaviour so I switched to enums, and then I realized I am not using them at all, so I just made a list of messages :)

I am currently using the preconditions class to:
- see if there's any
- print them if there's any and --build-profile is on / do non-incremental if there's any

Until I need to actually do something more involved with them I'll keep it simple.